travis-sonarcloud.sh

```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+if [ "$TRAVIS_EVENT_TYPE" == "pull_request" ]
+then
+  mvn --quiet jacoco:report jacoco:merge sonar:sonar -Preport-code-coverage -Dsonar.host.url=https://sonarcloud.io -Dsonar.organization=kiegroup -Dsonar.projectKey=org.optaplanner:optaplanner -Dsonar.login=$SONARCLOUD_TOKEN -Dsonar.pullrequest.base=$TRAVIS_BRANCH -Dsonar.pullrequest.branch=$TRAVIS_PULL_REQUEST_BRANCH -Dsonar.pullrequest.key=$TRAVIS_PULL_REQUEST -Dsonar.pullrequest.provider=GitHub -Dsonar.pullrequest.github.repository=$TRAVIS_PULL_REQUEST_SLUG
```

Comment:
I'd suggest using bash continuation lines to keep one parameter per line for better readability (and sorting the params alphabetically). You'll them be able to compare the parameter lists more easily visually. e.g.

```bash
mvn --quiet \
jacoco:report \
jacoco:merge \
sonar:sonar \
-Preport-code-coverage \
-Dsonar.host.url=https://sonarcloud.io \
-Dsonar.login=$SONARCLOUD_TOKEN \
-Dsonar.organization=kiegroup \
-Dsonar.projectKey=org.optaplanner:optaplanner \
-Dsonar.pullrequest.base=$TRAVIS_BRANCH \
-Dsonar.pullrequest.branch=$TRAVIS_PULL_REQUEST_BRANCH \ -Dsonar.pullrequest.key=$TRAVIS_PULL_REQUEST \
-Dsonar.pullrequest.github.repository=$TRAVIS_PULL_REQUEST_SLUG \
-Dsonar.pullrequest.provider=GitHub
```